File: WattApp/server/Server/DTOs/DeviceResponseDTO.cs
using Microsoft.AspNetCore.Routing.Constraints;
using System.Globalization;

namespace Server.DTOs
{
    public class DeviceResponseDTO
    {
        public long UserId { get; private set; }
        public string DeviceCategory { get; set; }
        public string DeviceType { get; set; }
        public string DeviceBrand { get;  set; }
        public string DeviceModel { get; set; }
        public string Name { get; private set; }
        public float EnergyInKwh { get; private set; }
        public float StandByKwh { get; private set; }
        public bool Visibility { get; private set; }
        public bool Controlability { get; private set; }
        public bool TurnOn { get; private set; }
    }
}

File: WattApp/client/api/models/DeviceResponseDTO.ts
/* istanbul ignore file */
/* tslint:disable */
/* eslint-disable */

export type DeviceResponseDTO = {
    Id?: number;
    UserId?: number;
    DeviceCategory?: string | null;
    DeviceType?: string | null;
    DeviceBrand?: string | null;
    DeviceModel?: string | null;
    readonly Name?: string | null;
    EnergyInKwh?: number;
    StandByKwh?: number;
    Visibility?: boolean;
    Controlability?: boolean;
    TurnOn?: boolean;
};
